airflow/contrib/operators/kubernetes_pod_operator.py

```diff
@@ -112,55 +113,54 @@ class KubernetesPodOperator(BaseOperator):  # pylint: disable=too-many-instance-
     """
     template_fields = ('cmds', 'arguments', 'env_vars', 'config_file')
 
+    @staticmethod
+    def create_labels_for_pod(context):
+        """
+        Generate labels for the pod s.t. we can track it in case of Operator crash
+        :param context:
+        :return:
+        """
+        labels = {
+            'dag_id': context['dag'].dag_id,
+            'task_id': context['task'].task_id,
+            'exec_date': context['ts']
```

Comment:
We should include the try_number in here too - (i.e if try 1 is still running somehow but we want to launch try 2 we don't want to monitor try 1 again.)

Reply:
Is that something that could happen? Would the execution date be the same?

Reply:
Yeah, if the task fails and it retires :) (It's unlikely to still be running, but some error/odd behavoiur could make it happen, but the "key" that uniqueue identifies a Task Instance is (dag_id,task_id, execution_date,try_number)

Reply:
Hi guys,

I think we should consider the desired behaviour with this.  If we're including the try_number then it opens a situation where multiple tries could be running alongside each other.  I don't think that is ever desired & would certainly cause problems with my jobs (back to the original cause of this one 😄).

If we want to avoid monitoring the previous try, then should we check for it & kill it if exists before starting the next?

Reply:
@danccooper @ashb Question about this, I think retry_number would actually be good here... Especially if you are debugging and [kubernetes][delete_worker_pods] is "False". Seems like it would delete the original pod under @danccooper's suggestion and you'd lose that for debugging purposes.

Reply:
On our k8s cluster, we keep pods in the namespace for some time no matter the state they finished in (success, OOMKilled, evicted,....) -- I do not maintain the cluster and the decision is not up to me. Not having `try_number` as an identifying label of a pod makes airflow fail to launch another attempt.

Reply:
On our k8s cluster, we keep pods in the namespace for some time no matter the state they finished in (success, OOMKilled, evicted,....) -- I do not maintain the cluster and the decision is not up to me. Not having try_number as an identifying label of a pod makes airflow fail to launch another attempt.

Reply:
Hey @dakov I was going to suggest you just set the reattach_on_restart flag to False, however from checking the code I'm not sure this will work as expected which is probably the behaviour you're reporting? After discussing wth @kaxil he has rasied https://github.com/apache/airflow/issues/10021 to cover this, suggest we continue discussion on there.